src/construct.ts

```diff
@@ -417,21 +422,13 @@ export class Node {
       throw new Error(`Cannot add children to "${this.path}" during synthesis`);
     }
 
-    if (childName in this._children) {
+    if (this._children[childName]) {
       const name = this.id ?? '';
       const typeName = this.host.constructor.name;
       throw new Error(`There is already a Construct with name '${childName}' in ${typeName}${name.length > 0 ? ' [' + name + ']' : ''}`);
     }
 
-    if (!childName && this.id) {
-      throw new Error(`cannot add a nameless construct to the named scope: ${this.path}`);
-    }
```

Comment:
I don't think this if branch can get called. Here's my reasoning:
1. `addChild` is only called if a non-null `scope` is passed to the `Node` constructor.
2. if a non-null `scope` is passed to the `Node` constructor, and `this.id` has a falsey value (like `undefined` or an empty string), then an error will be thrown in the `Node` constructor before `addChild` is called.
3. based on (1) and (2), when `addChild` is called, it's guaranteed that `childName` will not be a falsey value 
4. the if statement's body only executes if childName takes on a falsey value

I tried coming up with a test case that would actually throw this error but I don't think there's a way.